src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs

```diff
@@ -221,7 +221,7 @@ private static FormulaValue UntypedObjectArrayChecker(IRContext irContext, int i
                     return new BlankValue(irContext);
                 }
 
-                if (!(cov.Impl.Type is ExternalType et && et.Kind == ExternalTypeKind.Array))
+                if (!(cov.Impl.Type is ExternalType et && (et.Kind == ExternalTypeKind.Array || et.Kind == ExternalTypeKind.ArrayAndObject)))
```

Comment:
Consider methods on ExternalType `SupportsArrayAccess`, `SupportsFieldAccess` instead?

Reply:
Will do that in next iteration when we'll rework IUntypedObject